### ISSUE TITLE
Unstable Memory after upgrading from Triton 2.27 / nvidia 22.10

### ISSUE THREAD

**Issue by jturner116** [triton-inference-server/server#6051]
**Description**
Upgrading from 22.10, ORT models are consuming significantly more memory and running VRAM OOM.
**Triton Information**
What version of Triton are you using?
Upgraded to Triton 2.35.0 / Nvidia 23.06

Are you using the Triton container or did you build it yourself?
Triton container

**To Reproduce**
Example onnx file 
https://drive.google.com/file/d/1Hf0m6pR7PE4sfgK2keVQvWir6LRmB8TG/view?usp=sharing

Config
https://drive.google.com/file/d/1DuinwY_T1245zYyy1_j9YKf0I9Sal5j7/view?usp=sharing

Model loads, but inference requires far more memory than on 22.10 
Seems like the conv node is what is grabbing extra memory
```
onnx runtime error 6: Non-zero status code returned while running Conv node. Name:'/stages/stages.0/blocks/blocks.0/conv/mlp/fc1/Conv' Status Message: /workspace/onnxruntime/onnxruntime/core/providers/cuda/cuda_call.cc:121 std::conditional_t<THRW, void, onnxruntime::common::Status> onnxruntime::CudaCall(ERRTYPE, const char*, const char*, ERRTYPE, const char*, const char*, int) [with ERRTYPE = cudaError; bool THRW = true; std::conditional_t<THRW, void, onnxruntime::common::Status> = void] /workspace/onnxruntime/onnxruntime/core/providers/cuda/cuda_call.cc:114 std::conditional_t<THRW, void, onnxruntime::common::Status> onnxruntime::CudaCall(ERRTYPE, const char*, const char*, ERRTYPE, const char*, const char*, int) [with ERRTYPE = cudaError; bool THRW = true; std::conditional_t<THRW, void, onnxruntime::common::Status> = void] CUDA failure 2: out of memory ; GPU=0 ;
```

**Expected behavior**

I did not expect memory allocations to increase. I imagine ORT is to blame, but I haven't found any relevant issues there yet. I will keep looking. 

Thanks!
Jonah

**Comment by jturner116** [comment]
Here's an example showing the crazy memory grab

```
[StatusCode.INTERNAL] onnx runtime error 6: Non-zero status code returned while running Conv node. Name:'/stages/stages.3/blocks/blocks.0/conv/mlp/fc1/Conv' Status Message: /workspace/onnxruntime/onnxruntime/core/framework/bfc_arena.cc:368 void* onnxruntime::BFCArena::AllocateRawInternal(size_t, bool, onnxruntime::Stream*, bool, onnxruntime::WaitNotificationFn) Available memory of 1048576 is smaller than requested bytes of 140281169450752\n
```

I do not get the same memory complications running inference in ONNXRuntime 1.15.1 locally on GPU outside of Triton. I will try to reproduce locally on 1.15.0

**Comment by jturner116** [comment]
The issue seems async related. Here is the breaking inference pattern

```Python
        inputs = []
        inputs.append(grpcclient.InferInput('input', [len(batch), 3, 224, 224], "FP32"))
        inputs[0].set_data_from_numpy(batch)

        results = await asyncio.gather(
            *[client.infer(model['triton_model_name'], inputs=inputs) for model in self.models]
        )
```
where client is a import tritonclient.grpc.aio as grpcclient InferenceServerClient 
but sync requests to each model do not error OOM.

**Comment by Tabrizian** [comment]
This looks like a  change with onnxruntime memory allocations.

>I do not get the same memory complications running inference in ONNXRuntime 1.15.1 locally on GPU outside of Triton. I will try to reproduce locally on 1.15.0

When trying to reproduce this issue outside Triton, did you run multiple inferences? It sounds like you run into this issue only when you have multiple inflight requests.

There was a feature request for ORT backend to enable this that might be related to what you are running into: https://github.com/triton-inference-server/onnxruntime_backend/issues/166

**Comment by jturner116** [comment]
That is true, it was only related to inflight requests. I will try to test that more thoroughly outside of Triton.

That request is from January and does not explain the breaking change. My inference pattern stayed the same, but the OOM issue arose on Triton upgrade. It could be and is likely a breaking change within ONNXRuntime, but Triton should advertise that breaking change on the release.

**Comment by gilljon** [comment]
@drexalt Did you ever find a fix for this?

**Comment by drexalt** [comment]
@gilljon No, sorry!

**Comment by gilljon** [comment]
@drexalt So what was your workaround?

**Comment by drexalt** [comment]
@gilljon Ahh sorry I stopped working with it shortly after that, so I never successfully upgraded from the version I was on prior.
